pkg/auth/local_authenticator.go

```diff
@@ -0,0 +1,90 @@
+package auth
+
+import (
+	"crypto"
+	"net/http"
+
+	"github.com/dgrijalva/jwt-go"
+	"github.com/go-openapi/runtime"
+	"github.com/go-openapi/runtime/security"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+
+	"github.com/openshift/assisted-service/internal/common"
+	logutil "github.com/openshift/assisted-service/pkg/log"
+	"github.com/openshift/assisted-service/pkg/ocm"
+)
+
+type LocalAuthenticator struct {
+	log       logrus.FieldLogger
+	publicKey crypto.PublicKey
+}
+
+func NewLocalAuthenticator(cfg *Config, log logrus.FieldLogger) (*LocalAuthenticator, error) {
+	if cfg.ECPublicKey == "" {
+		return nil, errors.Errorf("local authentication requires an ecdsa Public Key")
+	}
+
+	key, err := jwt.ParseECPublicKeyFromPEM([]byte(cfg.ECPublicKey))
+	if err != nil {
+		return nil, err
+	}
+
+	a := &LocalAuthenticator{
+		log:       log,
+		publicKey: key,
+	}
+
+	return a, nil
+}
+
+func (a *LocalAuthenticator) AuthType() AuthType {
+	return TypeLocal
+}
+
+func (a *LocalAuthenticator) AuthAgentAuth(token string) (interface{}, error) {
+	_, err := validateToken(token, a.publicKey)
+	if err != nil {
+		a.log.Errorf("failed to validate token: %v", err)
+		return nil, err
+	}
+	return ocm.AdminPayload(), nil
+}
+
+func (a *LocalAuthenticator) AuthUserAuth(_ string) (interface{}, error) {
+	return nil, errors.Errorf("User Authentication not allowed for local auth")
+}
+
+func (a *LocalAuthenticator) CreateAuthenticator() func(_, _ string, authenticate security.TokenAuthentication) runtime.Authenticator {
+	return func(name string, _ string, authenticate security.TokenAuthentication) runtime.Authenticator {
+		return security.HttpAuthenticator(func(r *http.Request) (bool, interface{}, error) {
+			log := logutil.FromContext(r.Context(), a.log)
+
+			p, err := authenticate(r.Header.Get(name))
+			if err != nil {
+				log.Errorf("failed to authenticate: %v", err)
+				if common.IsKnownError(err) {
+					return true, nil, err
+				}
+				return true, nil, common.NewInfraError(http.StatusUnauthorized, err)
+			}
+			return true, p, nil
+		})
+	}
+}
+
+func validateToken(token string, pub crypto.PublicKey) (*jwt.Token, error) {
+	parsed, err := jwt.Parse(token, func(t *jwt.Token) (interface{}, error) { return pub, nil })
+
+	if err != nil {
+		return nil, errors.Errorf("Failed to parse token: %v\n", err)
+	}
+	if jwt.SigningMethodES256.Alg() != parsed.Header["alg"] {
```

Comment:
nit: Use the parsed `parsed.Method.Alg()` rather than the header map

[struct field definition ](https://github.com/dgrijalva/jwt-go/blob/317b82a6816a5f2b31ff8176cd6c7e5fe7564093/token.go#L25)